src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -579,6 +583,36 @@ public void Commit(IReleaseSpec releaseSpec, IWorldStateTracer stateTracer, bool
             {
                 ReportChanges(stateTracer, trace);
             }
+
+            if (commitRoots)
+            {
+                FlushToTree();
+            }
+        }
+
+        private void FlushToTree()
+        {
+            int writes = 0;
+            int skipped = 0;
+            foreach (var key in _blockChanges.Keys)
```

Comment:
Similar as above, why loop over to then get the value manually? I don't believe it's cheaper, the lookup hashes the key while the iterator over kvp just scans, right?

Reply:
Changed